src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1429,6 +1442,51 @@ protected MembersAndInitializers GetMembersAndInitializers()
             return membersAndInitializers!;
         }
 
+        /// <summary>
+        /// The purpose of this function is to assert that the <paramref name="member"/> symbol
+        /// is actually among the symbols cached by this type symbol in a way that ensures
+        /// that any consumer of standard APIs to get to type's members is going to get the same 
+        /// symbol (same instance) for the member rather than an equvalent, but different instance.
+        /// </summary>
+        [Conditional("DEBUG")]
+        internal void AssertMemberExposure(Symbol member, bool forDiagnostics = false)
+        {
+            if (member is NamedTypeSymbol type)
+            {
+                Debug.Assert(forDiagnostics);
+                Debug.Assert(Volatile.Read(ref _lazyTypeMembers)?.Values.Any(types => types.Contains(t => t == (object)type)) == true);
```

Comment:
>? [](start = 64, length = 1)

Since this assert expects `_lazyTypeMembers` to be non-null, you could just `!` away the nullable warning and simplify the condition.

Reply:
> Since this assert expects _lazyTypeMembers to be non-null, you could just ! away the nullable warning and simplify the condition.

I do not find this condition particularly complex and I didn't write it this way because a nullable warning. It naturally happened written this way. Also, it is not my goal here to assume something, and crash with null ref when the assumption is violated.  

---
In reply to: [571282589](https://github.com/dotnet/roslyn/pull/50961#discussion_r571282589) [](ancestors = 571282589)